src/analysis/special_functions/pow.lean

```diff
@@ -1137,6 +1128,18 @@ begin
   { exact coe_rpow_of_ne_zero hx _ }
 end
 
+lemma coe_rpow_def {p : nnreal × ℝ} :
+  (p.1 : ennreal) ^ p.2 = ite (p.1=0 ∧ p.2<0) ⊤ ↑(p.1 ^ p.2) :=
+begin
+  split_ifs,
+  { rw [h.left, ennreal.coe_zero, ennreal.zero_rpow_of_neg h.right], },
+  rw auto.not_and_eq at h,
+  cases h,
+    { rw coe_rpow_of_ne_zero h, },
+    { push_neg at h,
+      rw coe_rpow_of_nonneg _ h, },
+end
```

Comment:
This is even simpler:
```suggestion
lemma coe_rpow_def (x : ℝ≥0) (y : ℝ) :
  (x : ennreal) ^ y = if x = 0 ∧ y < 0 then ⊤ else (x ^ y : ℝ≥0) := rfl
```

Reply:
Well, I should have read the actual definition of rpow before writing a non rfl proof for a lemma I called def. Thanks.